cpp/src/replace/replace.cu

```diff
@@ -395,29 +645,90 @@ struct replace_nulls_column_kernel_forwarder {
                                                         *device_out,
                                                         valid_count);
 
-    cudaStreamSynchronize(stream);
-
     if (output_view.nullable()) {
-      output->set_null_count(output->size() - valid_counter.value());
+      output->set_null_count(output->size() - valid_counter.value(stream));
     }
 
     return output;
   }
+
   template<typename col_type,
       std::enable_if_t<not cudf::is_fixed_width<col_type>()>* = nullptr>
   std::unique_ptr<cudf::column> operator()(cudf::column_view const& input,
                                            cudf::column_view const& replacement,
                                            rmm::mr::device_memory_resource* mr,
                                            cudaStream_t stream = 0) {
-    CUDF_FAIL("Non-fixed-width types are not supported for replace.");
+    rmm::device_scalar<cudf::size_type> valid_counter(0, stream);
+    cudf::size_type *valid_count = valid_counter.data();
+
+    auto replace_first = replace_nulls_strings<0, false>;
+    auto replace_second = replace_nulls_strings<1, false>;
+    if (replacement.has_nulls()){
+      replace_first = replace_nulls_strings<0, true>;
+      replace_second = replace_nulls_strings<1, true>;
+    }
+
+    // Create new offsets column to use in kernel
+    std::unique_ptr<cudf::column> offsets = cudf::make_numeric_column(cudf::data_type(cudf::type_id::INT32),
+                                                                      input.size() + 1,
+                                                                      cudf::mask_state::UNALLOCATED,
+                                                                      stream,
+                                                                      mr);
+    auto offsets_view = offsets->mutable_view();
+    auto device_in = cudf::column_device_view::create(input);
+    auto device_replacement = cudf::column_device_view::create(replacement);
+
+    int valid_chars = cudf::bitmask_allocation_size_bytes(input.size());
+    rmm::device_buffer valid_bits(valid_chars, stream, mr);
+
+    // Call first pass kernel to get sizes in offsets
+    cudf::experimental::detail::grid_1d grid { input.size(), BLOCK_SIZE, 1 };
+    replace_first<<<grid.num_blocks, BLOCK_SIZE, 0, stream>>>(*device_in,
+                                                              *device_replacement,
+                                                              reinterpret_cast<cudf::bitmask_type*>(valid_bits.data()),
+                                                              offsets_view.data<cudf::size_type>(),
+                                                              nullptr,
+                                                              valid_count);
+
+    // Compute the offsets from the sizes
+    int32_t * offsets_data = offsets_view.data<int32_t>();
+    thrust::exclusive_scan(rmm::exec_policy(stream)->on(stream),
```

Comment:
`make_offsets_child_column` exists to abstract this stuff and prevent everyone from having to repeat it over and over again. https://github.com/rapidsai/cudf/blob/branch-0.11/cpp/include/cudf/strings/detail/utilities.cuh#L45

Reply:
Changed to use make_offsets_child_column.